packages/devp2p/src/ext/kbucket.ts

```diff
@@ -35,12 +35,14 @@ OTHER DEALINGS IN THE SOFTWARE.
 import { randomBytes } from '@ethereumjs/util'
 import { EventEmitter } from 'events'
 
+import type { KBucketOptions } from '../types'
+
 /**
  * @param  {Uint8Array} array1
  * @param  {Uint8Array} array2
  * @return {Boolean}
  */
-function arrayEquals(array1: any, array2: any) {
+function arrayEquals(array1: Uint8Array, array2: Uint8Array): boolean {
```

Comment:
Maybe this is a dumb question but how is this used? Is this used when we're trying to verify if two arrays point to the same underlying memory?

Reply:
This just seems like a re-implementation of equalsBytes. Refactored it!

Reply:
Okay, the code that was there certainly didn't implement true value level equality for arrays, and I'm on mobile so wasn't able to search and see how it's used.